include/graphblas/algorithms/amg/amg_data.hpp

```diff
@@ -0,0 +1,201 @@
+
+/*
+ *   Copyright 2021 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Data structures to store AMG input/output data.
+ * @file amg_data.hpp
+ * @author Alberto Scolari (alberto.scolari@huawei.com)
+ * @author Denis Jelovina (denis.jelovina@huawei.com)
+ * @date 2022-10-08
+ */
+
+#ifndef _H_GRB_ALGORITHMS_AMG_DATA
+#define _H_GRB_ALGORITHMS_AMG_DATA
+
+#include <vector>
+#include <cstddef>
+#include <graphblas.hpp>
+
+namespace grb {
+
+	namespace algorithms {
+
+		/**
+		 * basic data container for the AMG algorithm, storing \b only the
+		 * data in common between the full CG run and the V-cycle multi-grid solver.
+		 * Additional data are stored in inheriting data structures.
+		 *
+		 * @tparam IOType type of values of the vectors for intermediate results
+		 * @tparam NonzeroType type of the values stored inside the system matrix #A
+		 */
+		template< typename IOType, typename NonzeroType >
+		struct system_data {
+			const std::size_t system_size; ///< size of the system, i.e. side of the #A
```

Comment:
given that the comments below this one are all aligned, align this one too? (can be multi-line)